code/GameEngine/GameObject/GameObject.cs

```diff
@@ -232,6 +232,20 @@ public IEnumerable<T> GetComponents<T>( bool enabledOnly = true, bool deep = fal
 		}
 	}
 
+	public bool TryGetComponent<T>( out T component )
```

Comment:
We should have this on BaseComponent too

Reply:
pushed a fix for this, also added the parameters i missed